eng/pipelines/templates/Build_and_UnitTest.yml

```
@@ -43,6 +43,7 @@ steps:
   inputs:
     getDropNameByDrop: true
     optimizationInputsDropNamePrefix: OptimizationInputs/$(System.TeamProject)/$(Build.Repository.Name)
+    ShouldSkipOptimize: $(ShouldSkipOptimize)
```

Is there any downside to bootstrapping OptProf data? Does it take longer to train or something else?

---

"bootstrapping" is an expression for "get (thing) started".  When the OptProf installer can't find optimization input, the task fails, breaking the build. Setting this to `true` tells the task to ignore the error when the input data can't be found. When the current build finishes, OprProf training happens in a different pipeline, and then when a new build starts, it can download optimization data that now exists.

Normally we don't want the build to continue to when the input data can't be found, because it means that the build will be missing that optimization data, and therefore if we try to insert into VS, it will fail DDRITs. We only need it when a branch hasn't been built in such a long time that all the optimization data was deleted for being too old.

So if your question is why we don't always set this to true, that's why.